### DISCUSSION TITLE
Network share as local, showing 4TB free, 0B used

### DISCUSSION THREAD

**Discussion by minikold** [dokan-dev/dokany#982] [Q&A] [answered]
I have a few different network shares from an Unraid server that I mirror periodically for file transfer using Dokan. I use a simple command of:

`cd "C:\Program Files\Dokan\DokanLibrary-1.4.1\sample\mirror"
.\mirror.exe /r "\\servername\sharename" /l T`

I recently upgraded to 1.4.1 to try to resolve the issue, but the issue persists. 1.4.0 was functioning fine. The only change that transpired was that two of the shares that I map recently got upgraded storage to a 10TB drive from a 500GB drive. The 10TB shares are reporting as 3.99TB free, 0B used, but in actual are 4.33TB used, 5.67TB free. Whereas the other share that was on a 4TB drive directly (3.63TB actual), reports properly into Windows. 

The drives from Unraid:
![image](https://user-images.githubusercontent.com/9805585/113491302-5c080700-9495-11eb-880f-d66be5fd937a.png)

Windows File Explorer after Dokan mapping:
![image](https://user-images.githubusercontent.com/9805585/113491358-c15bf800-9495-11eb-988d-7838991221b4.png)

I'm curious if there is a 4TB limit by chance with this? The files are still accessible and writeable, so that's not an issue, but I would like them to report properly as that may affect another purpose that I utilize Dokan for.

I know the "xfs" filesystem is not an issue, as the drive that this 10TB replaced was the 500GB above it in the list, and that reports fine, as the "O:" drive in the screenshot.

**Comment by minikold** [comment]
I spoke too soon. Turns out it doesn't cause a problem with my other purpose for using Dokan. However, I would still like, if possible, for it to show the correct sizing. Not sure if that's a Dokan issue, or a Windows issue.

**Comment by Liryna** [comment, marked as answer]
I believe you are hitting the limit of `GetDiskFreeSpace` params (Congrats 😄 ).
https://github.com/dokan-dev/dokany/blob/master/samples/dokan_mirror/mirror.c#L1420-L1421
It should be changed for `GetDiskFreeSpaceEx`
https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getdiskfreespaceexa

**Reply by minikold** [reply]
Is that something I can switch in my mirror.c?

**Reply by Liryna** [reply]
Yes, patch you local mirror with https://github.com/dokan-dev/dokany/commit/1631cea6a5aead233ff588b31ef09c5baf3632e6
Do not build with the head library. It is not compatible with the driver of the last release.

**Reply by minikold** [reply]
I have edited my mirror.c with the changes listed on that commit, but it seems to be reporting the same way. I'm not sure what you mean by patching my mirror. (Sorry, I'm a novice to github.)

**Reply by Liryna** [reply]
After editing the file, have you rebuilt the binary ? You need visual studio to do that.

**Reply by minikold** [reply]
Sorry for a late reply, been a busy as heck week. I did not rebuild the binary. I do not know how to do that, I apologize.

**Reply by Liryna** [reply]
No problem, I will try to make a release with this patch in the coming days.

**Reply by lvincent81** [reply]
Any chance on getting that build this week Liryna?  No problem otherwise, but will save me try to figure out how to build this.

Thanks!

**Reply by Liryna** [reply]
No unfortunately, I am waiting to finish a patch before doing the release.
The process is long so I cannot do them that often.
That said, I should have the patch finished next week.
